druid-shell/src/platform/x11/application.rs

```diff
@@ -350,20 +361,71 @@ impl Application {
         Ok(false)
     }
 
-    pub fn run(self, _handler: Option<Box<dyn AppHandler>>) {
+    fn run_inner(self) -> Result<(), Error> {
+        // Try to figure out the refresh rate of the current screen. We run the idle loop at that
+        // rate.
+        let refresh_rate = util::refresh_rate(self.connection(), self.window_id).unwrap_or(60.0);
+        let timeout = Duration::from_millis((1000.0 / refresh_rate) as u64);
+        let mut last_idle_time = Instant::now();
         loop {
```

Comment:
I'd propose to add a call to `self.connection.flush()?` either at the end of the beginning of the loop. That way, you always flush before blocking in `poll` and thus all other calls to `flush` become unnecessary.